.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "deps/curl"]
+	path = deps/curl
+	url = git@github.com:curl/curl.git
```

Comment:
encounters issue with url form. besides I m unsure these deps handling is necessarily that appealing on unixes. why not using vcpkg for example for windows ?

Reply:
I agree, we don't want (nontrivial) dependencies in the repo, and we don't want to build them as part of the yq2 build.

Reply:
So, do you want users to specifically build and install curl? Isn't it less demanding to have dependency (as already is, because is being used in code) automatically build and compile instead of the opposite? How about debugging if, for example, there is an error that leads to curl?

Reply:
again package manager like conan/vcpkg are existing to make these sort of situations easier, especially if you plan to build several, for example, video games when you often see the same set of libraries ... sdl2/openal/opus etc ... you build/install once then that's it.